x-pack/plugins/enterprise_search/public/applications/analytics/components/analytics_collection_view/analytics_collection_data_view_logic.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { DataView } from '@kbn/data-views-plugin/common';
+
+import { KibanaLogic } from '../../../shared/kibana/kibana_logic';
+
+import { FetchAnalyticsCollectionLogic } from './fetch_analytics_collection_logic';
+
+interface AnalyticsCollectionDataViewLogicValues {
+  dataView: DataView | null;
+}
+
+interface AnalyticsCollectionDataViewLogicActions {
+  setDataView(dataView: DataView): { dataView: DataView };
+}
+
+export const AnalyticsCollectionDataViewLogic = kea<
+  MakeLogicType<AnalyticsCollectionDataViewLogicValues, AnalyticsCollectionDataViewLogicActions>
+>({
+  actions: {
+    setDataView: (dataView) => ({ dataView }),
+  },
+  listeners: ({ actions }) => ({
+    [FetchAnalyticsCollectionLogic.actionTypes.apiSuccess]: async (collection) => {
```

Comment:
nitpick: this should be explicitly connected with a `connect` property on the logic. We tend to do that to be more explicit, but also there's an issue in Kea where auto-connecting on listeners is flaky.

Reply:
Do you mean something like this?
`FetchAnalyticsCollectionLogic.connections['enterprise_search.api.analytics.analytics_collection_api_logic'].actionTypes.apiSuccess`